players/g6_player/__init__.py

```diff
@@ -25,48 +28,162 @@ def __init__(
         self.logger = logger
         self.maximum_door_frequency = maximum_door_frequency
         self.radius = radius
-        self.known_target = False
-        # Data structure to hold state information about the doors inside the radius
-        self.curr_maze = {}
+        self.maze_boundary_x = 201
+        self.maze_boundary_y = 201
+        # player_maze[x][y][cycle][door_direction] = boolean door open status
+        self.player_maze_freq = np.zeros((self.maze_boundary_x, self.maze_boundary_y, self.maximum_door_frequency, 4))
```

Comment:
so:
- 201 rows
- 201 columns
-- at this point we have each cell
- each cell is an array with a value for each of the possible door frequencies.
- each door frequency has four values

Could you clarify how we'll use this? 
My guess is that for each cell and tick [x][y][tick], we have an array to see which of the 4 doors in the cell are open for that tick: e.g. [1, 0, 0, 1] means [NORTH, ~~south~~, ~~east~~, WEST] are open.

Does that make sense?

Reply:
I am still not sure about how this is used, but I'd say we need to be wary of using the maximum_door_frequency variable, especially for the purpose of generating arrays, as there's no upper limit on this number.
Ken mentioned in class he could just as well use `maximum_door_frequency = 10.000`, that would make our array `201 x 201 x 10.000 x 4 = 1.6 billon` floats - thats 6GB for just this array.

Rather, we can just store a time the agent saw the door open two times to find the worst case frequency

Reply:
Robin and I spoke on Friday about this. Our strawman is that there may be some benefit to keeping track of the global coordinates, and since we don't know where the starting point is, we start at (100, 100) in a 201 x 201 grid, and adjust back to a 100 x 100 grid as soon as we find the borders. For example, if the real starting point is (18, 76), we would start at (100, 100), then after hitting two orthogonal borders, we would subtract all data points gathered thus far by (-82, -14), which would also adjust the original starting point to (18, 76).

In terms of memory, this is an O(1) increase from keeping just a 100 x 100 grid.  This question is whether there is a more efficient way to maintain the max_door_freq.

Reply:
I'm okay with having an array twice the size and starting in the middle. 
However, wouldn't we then use a 199x199 array instead and start at [99,99] (zero indexed)?